geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/query/QueryProcess.scala

```diff
@@ -98,6 +105,9 @@ class QueryVisitor(features: SimpleFeatureCollection,
     source.getFeatures(query)
   }
 
+  override def getExpressions: java.util.List[Expression] = {
+    FilterHelper.propertyNames(filter, origSft).map(ff.property).toList
```

Comment:
I think that would cause the sft of the feature collection to be modified as well... so they should match. But you're right that we should check for functions/transforms/etc - might need to call the query planner methods to get the return sft based on the properties - I think that would handle it...